source/plugins/ruby/kubelet_utils.rb

```diff
@@ -17,11 +18,14 @@ def get_node_capacity
         cpu_capacity = 1.0
         memory_capacity = 1.0
 
-        response = CAdvisorMetricsAPIClient.getNodeCapacityFromCAdvisor(winNode: nil)
+        response = CAdvisorMetricsAPIClient.getAllMetricsCAdvisor(winNode: nil)
         if !response.nil? && !response.body.nil?
-          cpu_capacity = JSON.parse(response.body)["num_cores"].nil? ? 1.0 : (JSON.parse(response.body)["num_cores"] * 1000.0)
-          memory_capacity = JSON.parse(response.body)["memory_capacity"].nil? ? 1.0 : JSON.parse(response.body)["memory_capacity"].to_f
-          @log.info "CPU = #{cpu_capacity}mc Memory = #{memory_capacity / 1024 / 1024}MB"
+          all_metrics = response.body.split("\n")
+          cpu_capacity = all_metrics.select{|m| m.start_with?('machine_cpu_cores') }.first.split.last.to_f * 1000
+          @log.info "CPU Capacity #{cpu_capacity}"
+          memory_capacity_e = all_metrics.select{|m| m.start_with?('machine_memory_bytes') }.first.split.last
```

Comment:
as vishwa commented in other comment, having strict check on metric name will ensure we are picking the right metric.